packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/project/data/versioning/services/DefaultProjectDataVersiongRestService.java

```diff
@@ -0,0 +1,129 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.project.data.versioning.services;
+
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IObjectService;
+import org.eclipse.sirius.web.application.dto.Identified;
+import org.eclipse.sirius.web.application.object.services.api.IDefaultObjectRestService;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.ChangeType;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.RestCommit;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.RestDataIdentity;
+import org.eclipse.sirius.web.application.project.data.versioning.dto.RestDataVersion;
+import org.eclipse.sirius.web.application.project.data.versioning.services.api.IDefaultProjectDataVersioningRestService;
+import org.springframework.stereotype.Service;
+
+/**
+ * Implementation of {@link IDefaultProjectDataVersioningRestService}.
+ *
+ * @author arichard
+ */
+@Service
+public class DefaultProjectDataVersiongRestService implements IDefaultProjectDataVersioningRestService {
```

Comment:
I don't mind anything specific in this class but I wonder, are you sure that you will need to access the live up-to-date version of the semantic data for all these operations?

It seems that all those operations can be done using the snapshot of the semantic data from the database. I thus don't think any of those methods need the editing context. As a result, you would not need any internal input / payload / event handler / etc. You would instead just call those methods from your controller. If you just need the id of the project, you have that from the controller.

Reply:
Ok, so do you will merge it as it is or do I need to do something? I don't have a strong opinion, if you think it's better to use the snapshot of the semantic data from the database, I can update the code.